Frameworks/BlockchainCertificates/Certificate.swift

```diff
@@ -462,6 +480,124 @@ private enum MethodsForV1_2 {
     }
 }
 
+// MARK: Certificate Version 2.0
+private enum MethodsForV2 {
+    
+    static func parse(issuerJSON: AnyObject?) -> Issuer? {
+        guard let issuerData = issuerJSON as? [String : String],
+            let issuerURLString = issuerData["url"],
+            let issuerURL = URL(string: issuerURLString),
+            let logoURI = issuerData["image"],
+            let issuerEmail = issuerData["email"],
+            let issuerName = issuerData["name"],
+            let issuerId = issuerData["id"],
+            let issuerIdURL = URL(string: issuerId) else {
+                return nil
+        }
+        let logo = imageData(from: logoURI)
+        var revocationURL : URL? = nil
+        
+        if issuerData["revocationList"] != nil {
+            revocationURL = URL(string: issuerData["revocationList"]!)
+        }
+
+        return Issuer(name: issuerName,
+                      email: issuerEmail,
+                      image: logo,
+                      id: issuerIdURL,
+                      url: issuerURL,
+                      revocationURL: revocationURL)
+    }
+    
+    static func parse(recipientJSON: AnyObject?) -> Recipient? {
+        guard let recipientData = recipientJSON as? [String : AnyObject],
+            let recipientProfile = recipientData["recipientProfile"] as? [String : AnyObject],
+            let name = recipientProfile["name"] as? String, // difference from v1.2
```

Comment:
Thanks for noting the differences. I was worried this was all copy/pasted from v1.2